src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -2,38 +2,51 @@ using Oceananigans.Advection:
     _advective_momentum_flux_Uu,
     _advective_momentum_flux_Uv,
     _advective_momentum_flux_Vu,
-    _advective_momentum_flux_Vv
+    _advective_momentum_flux_Vv,
+    vertical_vorticity_U,
+    vertical_vorticity_V,
+    bernoulli_head_U,
+    bernoulli_head_V
 
 using Oceananigans.Grids: AbstractGrid
 using Oceananigans.Operators: Ax_qᶠᶜᶜ, Ay_qᶜᶠᶜ
 
+
 #####
 ##### Momentum flux operators
 #####
 
-@inline momentum_flux_huu(i, j, k, grid, advection, solution) =
+@inline momentum_flux_huu(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
 
-@inline momentum_flux_hvu(i, j, k, grid, advection, solution) =
+@inline momentum_flux_hvu(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Vu(i, j, k, grid, advection, solution.vh, solution.uh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
-@inline momentum_flux_huv(i, j, k, grid, advection, solution) =
+@inline momentum_flux_huv(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Uv(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
-@inline momentum_flux_hvv(i, j, k, grid, advection, solution) =
+@inline momentum_flux_hvv(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Vv(i, j, k, grid, advection, solution.vh, solution.vh) / solution.h[i, j, k]
 
 #####
 ##### Momentum flux divergence operators
 #####
 
-@inline div_hUu(i, j, k, grid, advection, solution) =
-    1 / Vᶠᶜᶜ(i, j, k, grid) * (δxᶠᵃᵃ(i, j, k, grid, momentum_flux_huu, advection, solution) +
-                               δyᵃᶜᵃ(i, j, k, grid, momentum_flux_hvu, advection, solution))
+@inline div_hUu(i, j, k, grid, advection, solution, ::VectorInvariantFormulation) = (
+    + vertical_vorticity_U(i, j, k, grid, advection, solution.uh, solution.vh)  # Vertical relative vorticity term
```

Comment:
I gather that `vertical_vorticity_U` and `vertical_vorticity_V` is the same field but at two different locations?  I presume this is what is done in HydrostaticModel so no problem but I would have thought that using `fcc` and `cfc` might be appropriate?

Reply:
Nothing worth changing here.  Just a thought.